lib/scenario_runner.py

```diff
@@ -257,22 +257,35 @@ def _join_paths(self, path, path2, force_path_as_root=False):
 
         raise FileNotFoundError(f"{path2} in {path} not found")
 
-    def _validate_image_architecture(self, image_name: str) -> None:
-        """Validate that the pulled Docker image architecture is compatible with the host."""
+    def _validate_image_architecture(self, image_name: str, raise_on_mismatch: bool = True) -> tuple[str, str, bool]:
+        """Validate that the pulled Docker image architecture is compatible with the host.
+
+        Args:
+            image_name: The Docker image name to check
+            raise_on_mismatch: If True, raises RuntimeError on architecture mismatch
+
+        Returns:
+            Tuple of (image_arch, host_arch, is_compatible)
+        """
         ps = subprocess.run(
             ['docker', 'image', 'inspect', image_name, '--format', '{{.Architecture}}'],
             stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='UTF-8', check=False
         )
 
         if ps.returncode != 0:
-            raise RuntimeError(f"Failed to inspect Docker image architecture for '{image_name}': {ps.stderr}")
+            if raise_on_mismatch:
+                raise RuntimeError(f"Failed to inspect Docker image architecture for '{image_name}': {ps.stderr}")
+            return "", "", False
 
         image_arch = ps.stdout.strip()
         normalized_host_arch = map_host_to_docker_arch()
+        is_compatible = image_arch == normalized_host_arch
 
-        if image_arch != normalized_host_arch:
+        if not is_compatible and raise_on_mismatch:
```

Comment:
I see that here should be no raise, as the check is done later. But should this line not fully removed if we never set raise_on_mismatch = true?